### PR TITLE
Refs [Task][41620] Fetch and display data: Result List Screen

### PR DESCRIPTION
**Pull request by lanch1147**
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/41620?issue_count=21&issue_position=12&next_issue_id=41619&prev_issue_id=41622

#### Screenshots (nếu có)
https://user-images.githubusercontent.com/86757854/134641422-84d30c86-376d-4984-9933-f2132d6a6257.mov

### CODE REVIEW

File: Recipe Search/Recipe Search/Models/API Services/Request/RecipeRequest.swift [hunk 1]
```diff
@@ -49,6 +53,12 @@ struct RecipeRequest: APIRequest {
         self.selectedFields = selectedFields
     }
     
+    init(completeURL: URL) {
+        self.completeURL = completeURL
+        self.categories = []
+        self.selectedFields = []
```
Comment (tuanlvt-1117) [inline Recipe Search/Recipe Search/Models/API Services/Request/RecipeRequest.swift:59]:
```suggestion
       categories = []
       selectedFields = []
```

File: Recipe Search/Recipe Search/Controllers/ResultViewController.swift [hunk 2]
```diff
@@ -21,5 +26,70 @@ final class ResultViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
+        setupCollectionView()
+        fetchNextPage(with: initialRequest)
+    }
+    
+    // MARK: - Setup Collection View
+    
+    func setupCollectionView() {
```
Comment (tuanlvt-1117) [inline Recipe Search/Recipe Search/Controllers/ResultViewController.swift:35]:
```suggestion
    }
    
    func setupCollectionView() {
```

File: Recipe Search/Recipe Search/Controllers/ResultViewController.swift [hunk 3]
```diff
@@ -8,10 +8,15 @@
 import UIKit
 
 final class ResultViewController: UIViewController {
-    private var recipeResponse: RecipeResponse
+    @IBOutlet weak var collectionView: UICollectionView!
```
Comment (namtn-2711) [inline Recipe Search/Recipe Search/Controllers/ResultViewController.swift:11]: private

File: Recipe Search/Recipe Search/Controllers/ResultViewController.swift [hunk 4]
```diff
@@ -21,5 +26,70 @@ final class ResultViewController: UIViewController {
...
+        let layout = UICollectionViewCompositionalLayout(section: section)
+        return layout
+    }
+}
+
+extension ResultViewController: UICollectionViewDataSource {
+    private func fetchNextPage(with request: RecipeRequest) {
+        isFetching = true
+        APIFetcher.shared.fetchData(with: request) { [weak self] (result) in
+            self?.isFetching = false
```
Comment (namtn-2711) [inline Recipe Search/Recipe Search/Controllers/ResultViewController.swift:63]: unwrap self

File: Recipe Search/Recipe Search/Controllers/ResultViewController.swift [hunk 5]
```diff
@@ -21,5 +26,69 @@ final class ResultViewController: UIViewController {
...
+    func setupCollectionView() {
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        collectionView.registerCell(ofType: RecipeCollectionViewCell.self)
+        collectionView.setCollectionViewLayout(generateLayout(), animated: true)
+    }
+    
+    // MARK: - Compostional Layout
+    
+    func generateLayout() -> UICollectionViewCompositionalLayout {
```
Comment (tuanlvt-1117) [inline Recipe Search/Recipe Search/Controllers/ResultViewController.swift:42]:
```suggestion
    }
    
    func generateLayout() -> UICollectionViewCompositionalLayout {
```

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**